rtmessage: check for null pointer in rtMessageHeader_Decode

Permanence-AI-Coder opened the pull request:
The changes in this PR were automatically generated by the Permanence AI Coder and reviewed by @jweese and @dnovick.

This change adds explicit null checks for `hdr` and `ptr` to prevent potential null pointer dereferences inside the `rtEncoder_Decode` calls. 

This PR complies with RDK LLM usage requirements.

on-behalf-of: @permanence-ai <github-ai@permanence.ai>

karuna2git commented on src/rtmessage/rtMessageHeader.c:
```diff
@@ -86,7 +86,17 @@ rtMessageHeader_Encode(rtMessageHeader* hdr, uint8_t* buff)
 rtError
 rtMessageHeader_Decode(rtMessageHeader* hdr, uint8_t const* buff)
 {
+  if(hdr == NULL)
```
Can you use `RT_CHECK_INVALID_ARG` to return `RT_ERROR_INVALID_ARG`.

jweese replied:
Done.

karuna2git commented on src/rtmessage/rtMessageHeader.c:
```diff
@@ -86,7 +86,17 @@ rtMessageHeader_Encode(rtMessageHeader* hdr, uint8_t* buff)
+    return RT_ERROR;
+  }
   uint8_t const* ptr = buff;
+  if(ptr == NULL)
```
Can you use `RT_CHECK_INVALID_ARG` to return `RT_ERROR_INVALID_ARG`.

karuna2git commented on src/rtmessage/rtMessageHeader.c:
As you are adding the change in the beginning of the function appropriately, Can you remove this also.
Looks like the NULL pointer check is added after using the pointer.

jweese replied:
We removed this one, and the second check below that also looks redundant for the same reason.

Permanence-AI-Coder committed:
rtmessage: add null checks for arguments in rtMessageHeader_Decode

The changes in this PR were automatically generated by the Permanence AI
Coder and reviewed by @jweese and @dnovick. This change adds null checks
for the `hdr` and `buff` arguments at the start of
`rtMessageHeader_Decode`. It also removes redundant null checks for
`ptr` later in the function.

on-behalf-of: @permanence-ai <github-ai@permanence.ai>

karuna2git committed:
Merge branch 'develop' into permai-rtmessage-header-nullptr

karuna2git committed:
Merge branch 'develop' into permai-rtmessage-header-nullptr
